streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -227,7 +227,7 @@ public void writeAsText(String filePath, String streamName, Serde<K> keySerde, S
         try {
             PrintWriter printWriter = null;
             printWriter = new PrintWriter(filePath, StandardCharsets.UTF_8.name());
-            topology.addProcessor(name, new KeyValuePrinter<>(printWriter, keySerde, valSerde, streamName), this.name);
+            topology.addProcessor(name, new KStreamPeek<>(printWriter, keySerde, valSerde, streamName), this.name);
```

Comment:
It would be cleaner to call this via
```
new KStreamPeek<>(new PrintForeachAction(printWriter, keySerde, valSerde, streamName), false)
```

cf. comment below about `KStreamPeekPrinterProcessor`